specification/search/data-plane/Azure.Search/preview/2025-08-01-preview/knowledgeagent.json

```diff
@@ -0,0 +1,529 @@
+          "KnowledgeAgentRetrieve": {
+            "$ref": "./examples/KnowledgeAgentRetrieve.json"
+          }
+        },
+        "description": "KnowledgeAgent retrieves relevant data from backing stores.",
+        "parameters": [
+          {
+            "name": "retrievalRequest",
+            "in": "body",
+            "required": true,
+            "schema": {
+              "$ref": "#/definitions/KnowledgeAgentRetrievalRequest"
+            },
+            "description": "The retrieval request to process."
+          },
+          {
+            "$ref": "#/parameters/ClientRequestIdParameter"
+          },
+          {
+            "$ref": "#/parameters/ApiVersionParameter"
+          },
+          {
+            "$ref": "#/parameters/QuerySourceAuthorizationParameter"
+          }
+        ],
+        "x-ms-request-id": "request-id",
+        "responses": {
+          "200": {
+            "description": "The retrieval response is successfully returned.",
+            "schema": {
+              "$ref": "#/definitions/KnowledgeAgentRetrievalResponse"
+            }
+          },
+          "default": {
+            "description": "Error response.",
+            "schema": {
+              "$ref": "../../../../../common-types/data-plane/v1/types.json#/definitions/ErrorResponse"
+            }
+          }
+        }
+      }
+    }
+  },
+  "definitions": {
+    "KnowledgeAgentRetrievalRequest": {
+      "type": "object",
+      "properties": {
+        "messages": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KnowledgeAgentMessage"
+          }
+        },
+        "knowledgeSourceParams": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KnowledgeSourceParams"
+          }
+        }
+      },
+      "required": [
+        "messages"
+      ],
+      "description": "The input contract for the retrieval request."
+    },
+    "KnowledgeAgentRetrievalResponse": {
+      "type": "object",
+      "properties": {
+        "response": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KnowledgeAgentMessage"
+          }
+        },
+        "activity": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KnowledgeAgentActivityRecord"
+          },
+          "description": "The activity records for tracking progress and billing implications."
+        },
+        "references": {
+          "type": "array",
+          "items": {
+            "$ref": "#/definitions/KnowledgeAgentReference"
+          },
+          "description": "The references for the retrieval data used in the response."
+        }
+      },
+      "description": "The output contract for the retrieval response."
+    },
+    "KnowledgeSourceParams": {
+      "type": "object",
+      "discriminator": "kind",
+      "properties": {
+        "kind": {
+          "$ref": "searchservice.json#/definitions/KnowledgeSourceKind",
```

Comment:
The way the code generators work may result in this enum being generated multiple times. Is that okay for the SDK API?

That aside, this design is good to me from a Swagger perspective, no real need to change this. I said the above more for a future reference.